src/api/controllers/PostController.ts

```diff
@@ -0,0 +1,60 @@
+import {
+    JsonController, Params, Get, Post, Delete, UseBefore, Body, BodyParam,
+  } from 'routing-controllers';
+  import { UserModel } from '../../models/UserModel'
+  import { PostModel } from '../../models/PostModel'
+  import { PostService } from '../../services/PostService';
+  import { Inject } from 'typedi';
+  import {
+    GetPostsResponse,
+    GetPostResponse,
+    CreatePostRequest,
+    Uuid,
+    GenericSuccessResponse,
+  } from '../../types';
+  import { EmailParam, UuidParam } from '../validators/GenericRequests';
+  
+  // import { UserAuthentication } from '../middleware/UserAuthentication';
+  // import { AuthenticatedUser } from '../decorators/AuthenticatedUser';
+  
+  // @UseBefore(UserAuthentication)
+  @JsonController('post/')
+  export class PostController {
+    private postService: PostService;
+  
+    constructor(postService: PostService) {
+      this.postService = postService;
+    }
+  
+    // @Get()
+    // async helloWorldName(@BodyParam("name") name: string): Promise<string> {
+    //   return "Hello world, ".concat(name, "!");
+    // }
+  
+    @Get()
+    async getPosts(): Promise<GetPostsResponse> {
+      const posts = await this.postService.getAllPosts();
+      return { error: null, posts: posts.map((post) => post.getPostInfo()) };
+    }
+  
+    @Get('id/:id/')
```

Comment:
this isn't consistent with how youre doing `userId` later... shouldn't this be `postId` to keep naming consistent?